### PR TITLE
feat: #1638 Forest Client Search integration and show FC client name

### PR DESCRIPTION
**Pull request by ianliuwk1019**
Ref: #1638 
- Add `search` function to forest_client_integration.py to integrate with FC API Search endpoint.
- Create a `post_sync_forest_clients_dec` decorator to update get_application_role_assignments function return results 
  client_name (if available) through FC API Search endpoint.
- Adjust better naming for model's "client_number" relationship to "forest_client_relation" and adjust schema and other
  code impacted by the change.
- Client Code gen for the change of "client_number" object name to "forest_client".
- Adjust frontend ManagePermissionTable to display client name using `formatForestClientDisplayName`.
- Adjust exisiting `test_get_application_role_assignments_on_pagination` so tests can be tested without decorator (in isolation).
- Add tests for FC API Search integration testing.
- Add tests for new `post_sync_forest_clients_dec` decorator.

### CODE REVIEW

File: frontend/src/views/ManagePermissionsView/utils.ts [hunk 1]
```diff
@@ -44,11 +37,7 @@ export const generateAppPermissionSuccessNotifications = (
             .map((filtered) => filtered.detail.role);
 
     const successClientList = successRoleList
-        .map((role) =>
-            isFamRoleWithClientSchema(role)
-                ? role.forest_client
-                : role.client_number
-        )
```
Comment (ianliuwk1019) [inline frontend/src/views/ManagePermissionsView/utils.ts:51]: Backend has changed to use `role.forest_client` for consistency, so no need to use `role.client_number` anymore (did not make sense for the previous name)

File: server/admin_management/api/app/models/model.py [hunk 2]
```diff
@@ -245,7 +245,7 @@ class FamRole(Base):
 
     application: Mapped[FamApplication] = relationship("FamApplication", back_populates="fam_role")
 
-    client_number: Mapped['FamForestClient'] = relationship(
+    forest_client_relation: Mapped['FamForestClient'] = relationship(
```
Comment (ianliuwk1019) [inline server/admin_management/api/app/models/model.py:248]: Minor variable refactoring on naming that make sense.

File: server/backend/api/app/models/model.py [hunk 3]
```diff
@@ -617,7 +617,7 @@ class FamRole(Base):
     application: Mapped[FamApplication] = relationship(
         "FamApplication", back_populates="fam_role"
     )
-    client_number: Mapped[FamForestClient] = relationship(
+    forest_client_relation: Mapped[FamForestClient] = relationship(
```
Comment (ianliuwk1019) [inline server/backend/api/app/models/model.py:620]: Minor variable refactoring on naming that make sense.

File: frontend/src/components/ManagePermissionsTable/index.vue [hunk 4]
```diff
@@ -448,6 +449,33 @@ const highlightNewUserAccessRow = (
...
+                  forestClientData.forest_client_number,
+                  forestClientData.client_name
+              )
+            : "";
+    } else {
+        // For delegated admin data.
+        // TODO: No client name available from backend yet, implement soon. Only display client number. # noqa NOSONAR
+        return forestClientData?.forest_client_number;
+    }
+};
```
Comment (craigyu) [inline frontend/src/components/ManagePermissionsTable/index.vue:478]: Could you move this to the utils.ts under the same folder? so the file doesn't get too long

and perhaps rename it to getOrganizationName?

**Review by craigyu — CHANGES_REQUESTED** [review]
Looks great Ian! Added one request.

Also let's change the `_` to `-` in the endpoint while we're add it :) 
image

File: frontend/src/components/ManagePermissionsTable/index.vue (hunk 4 above)
Comment (ianliuwk1019) [reply frontend/src/components/ManagePermissionsTable/index.vue]: Thank for the suggestion Craig! This is moved.

**Comment by ianliuwk1019** [comment]
> Looks great Ian! Added one request.
> 
> Also let's change the `_` to `-` in the endpoint while we're add it :) image

Hi Craig, for the swagger's 'search' endpoint, I have add this change of "_" to "-" in the ticket #1639 to do all the endpoints once and for all. Also that ticket will verify and swap internal calling to replace `find_by_client_number` if that's reasonable.

**Review by craigyu — APPROVED** [review]

**Pull request merged**